Test.java

```diff
@@ -0,0 +1,12 @@
+class MyClass {
+  MyClass(MyClass mc) { }
+  
+  int     foo1() { return 0; }
+  void    foo2(int value) { }
+  int     foo3(int value) { return 0; } // Noncompliant
+  Object  foo4(int value) { return null; }
+  MyClass foo5(MyClass value) {return null; } // Noncompliant
+  
+  int     foo6(int value, String name) { return 0; }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove these unused method parameters. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1172)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS3400)